src/products/components/ProductVariantImageSelectDialog/ProductVariantMediaSelectDialog.tsx

```diff
@@ -48,43 +50,54 @@ const useStyles = makeStyles(
 );
 
 interface ProductVariantImageSelectDialogProps {
-  images?: ProductImageFragment[];
-  selectedImages?: string[];
+  media?: ProductMediaFragment[];
+  selectedMedia?: string[];
   open: boolean;
   onClose();
-  onImageSelect(id: string);
+  onMediaSelect(id: string);
 }
 
-const ProductVariantImageSelectDialog: React.FC<ProductVariantImageSelectDialogProps> = props => {
-  const { images, open, selectedImages, onClose, onImageSelect } = props;
-
+const ProductVariantMediaSelectDialog: React.FC<ProductVariantImageSelectDialogProps> = props => {
+  const { media, open, selectedMedia, onClose, onMediaSelect } = props;
   const classes = useStyles(props);
 
   return (
     <Dialog onClose={onClose} open={open}>
       <DialogTitle>
         <FormattedMessage
-          defaultMessage="Image Selection"
+          defaultMessage="Media Selection"
           description="dialog header"
         />
       </DialogTitle>
       <DialogContent>
         <div className={classes.root}>
-          {images
+          {media
             .sort((prev, next) => (prev.sortOrder > next.sortOrder ? 1 : -1))
-            .map(tile => (
+            .map(mediaObj => (
               <div
                 className={classNames([
                   classes.imageContainer,
                   {
                     [classes.selectedImageContainer]:
-                      selectedImages.indexOf(tile.id) !== -1
+                      selectedMedia.indexOf(mediaObj.id) !== -1
                   }
                 ])}
-                onClick={onImageSelect(tile.id)}
-                key={tile.id}
+                onClick={onMediaSelect(mediaObj.id)}
+                key={mediaObj.id}
               >
-                <img className={classes.image} src={tile.url} />
+                {mediaObj?.type !== ProductMediaType.IMAGE ? (
+                  <ProductMediaVideo
```

Comment:
Arent we doing this condition: either productMediaVideo or img quite a few times? Wouldn't a component be suitable instead of rewrites?